### ISSUE TITLE
what is the loss value when you start training?

### ISSUE THREAD

**Issue by codelvin** [JT-Ushio/MHA2MLA#12]
When I start mha-mla training, loss value goes up to 14-16, even though it drops very quickly to around 2.4 within 0.5B tokens. the starting loss value is not included in the paper, so I wonder what is the begining loss? Thanks a lot!

**Comment by JT-Ushio** [comment]
> When I start mha-mla training, loss value goes up to 14-16, even though it drops very quickly to around 2.4 within 0.5B tokens. the starting loss value is not included in the paper, so I wonder what is the begining loss? Thanks a lot!


Hi @codelvin, the high starting loss value is expected. Architectural modifications like **partial-RoPE** and **low-rank approximations** introduce an initial mismatch (e.g., pre-trained parameters in `layer_norm` layers may no longer be optimal for the modified model). As a result, the model requires a small amount of fine-tuning to quickly adapt.

For reference, here is the full loss curve of the SLM-135M-d_kv_8 model during pretraining.

![Image](https://github.com/user-attachments/assets/e8066a46-a5d1-49d6-806e-56dbc9744e43)

**Comment by codelvin** [comment]
Yeah I think around 12 is normal, because training any models from scratch has a starting loss around 12. 14-16 seems strange, because it cannot be higher than training from scratch. let me figure it out anyway. You can mark this as completed . Thanks!
